IPython/core/inputtransformer.py

```diff
@@ -1,16 +1,18 @@
-"""Input transformer classes to support IPython special syntax.
+"""DEPRECATED: Input transformer classes to support IPython special syntax.
+
+This module was deprecated in IPython 7.0, in favour of inputtransformer2.
 
 This includes the machinery to recognise and transform ``%magic`` commands,
 ``!system`` commands, ``help?`` querying, prompt stripping, and so forth.
 """
 import abc
 import functools
 import re
+import tokenize
+from tokenize import generate_tokens, untokenize, TokenError
```

Comment:
generate_token is deprecated/not public. Maybe figure out how to use tokenize directly ? Or complainc to CPython to undeprecate it cause it is annoying to use.

Reply:
I did ask them to make it official: https://bugs.python.org/issue12486#msg313591

It's been there unchanged for quite a few releases, so I doubt it's in imminent danger of removal. I'll try opening a PR to make it public.

Reply:
My PR was accepted, so `generate_tokens` is documented for Python 3.8. https://docs.python.org/3.8/library/tokenize.html#tokenize.generate_tokens